src/Persistence/Reflection/EnumReflectionProperty.php

```diff
@@ -86,13 +86,25 @@ private function fromEnum($enum)
     }
 
     /**
-     * @param int|string|int[]|string[] $value
+     * @param int|string|int[]|string[]|BackedEnum|BackedEnum[] $value
      *
      * @return ($value is int|string ? BackedEnum : BackedEnum[])
      */
     private function toEnum($value)
     {
+        if ($value instanceof BackedEnum) {
+            return $value;
+        }
+
         if (is_array($value)) {
+            foreach ($value as $v) {
```

Comment:
This was hard to read for me (maybe because I'm in vacations-mode ;)... IMO using `reset` would be more elegant than this entire loop

Reply:
Enjoy your vacations!